internal/auth0/organization/resource_connections.go

```diff
@@ -0,0 +1,230 @@
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/auth0/terraform-provider-auth0/internal/config"
+	"github.com/auth0/terraform-provider-auth0/internal/value"
+)
+
+// NewConnectionsResource will return a new auth0_organization_connections (1:many) resource.
+func NewConnectionsResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"organization_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "ID of the organization on which to enable the connections.",
+			},
+			"enabled_connections": {
+				Type: schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"connection_id": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The ID of the connection to enable for the organization.",
+						},
+						"assign_membership_on_login": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+							Description: "When true, all users that log in with this connection will be " +
+								"automatically granted membership in the organization. When false, users must be " +
+								"granted membership in the organization before logging in with this connection.",
+						},
+					},
+				},
+				Required:    true,
+				Description: "IDs of the connections that are enabled for the organization.",
+			},
+		},
+		CreateContext: createOrganizationConnections,
+		ReadContext:   readOrganizationConnections,
+		UpdateContext: updateOrganizationConnections,
+		DeleteContext: deleteOrganizationConnections,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Description: "With this resource, you can manage enabled connections on an organization.",
+	}
+}
+
+func createOrganizationConnections(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+	mutex := meta.(*config.Config).GetMutex()
+
+	organizationID := data.Get("organization_id").(string)
+
+	mutex.Lock(organizationID)
+	defer mutex.Unlock(organizationID)
+
+	alreadyEnabledConnections, err := api.Organization.Connections(organizationID)
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+
+		return diag.FromErr(err)
+	}
+
+	data.SetId(organizationID)
+
+	connectionsToAdd := expandOrganizationConnections(data.GetRawConfig().GetAttr("enabled_connections"))
+
+	if diagnostics := guardAgainstErasingUnwantedConnections(
+		organizationID,
+		alreadyEnabledConnections.OrganizationConnections,
+		connectionsToAdd,
+	); diagnostics.HasError() {
+		data.SetId("")
+		return diagnostics
+	}
```

Comment:
We need to do this because Adding a connection that is already there will throw an error, it's not like the assignRole or assignPermission behavior. We're also doing this check for most other 1:many resources as well that aren't roles or permissions.